src/commands/up.ts

```diff
@@ -1,51 +1,75 @@
 import cliSpinners from 'cli-spinners';
 import { existsSync } from 'std/fs/exists.ts';
 import * as path from 'std/path/mod.ts';
+import { animals, uniqueNamesGenerator } from 'unique-names-generator';
 import winston, { Logger } from 'winston';
 import { CloudGraph } from '../cloud-graph/index.ts';
-import { parseEnvironment } from '../environments/index.ts';
+import { DatacenterRecord } from '../datacenters/store.ts';
+import { Environment, parseEnvironment } from '../environments/index.ts';
 import { ImageRepository } from '../oci/index.ts';
 import { Pipeline, PlanContextLevel } from '../pipeline/index.ts';
 import { BaseCommand, CommandHelper, GlobalOptions } from './base-command.ts';
 import { destroyEnvironment } from './destroy/environment.ts';
 import { streamLogs } from './logs.ts';
 
 type UpOptions = GlobalOptions & {
-  datacenter: string;
-  environment: string;
   verbose: boolean;
   debug: boolean;
   ingress?: string[];
-};
+} & ({ environment: string; datacenter?: never } | { datacenter: string; environment?: never });
 
 const UpCommand = BaseCommand()
   .description('Spin up an environment that will clean itself up when you terminate the process')
   .arguments('[...components:string]')
-  .option('-d, --datacenter <datacenter:string>', 'The datacenter to use for the environment', { required: true })
-  .option('-e, --environment <environment:string>', 'The name of your tmp environment', { default: 'local' })
+  .option('-d, --datacenter <datacenter:string>', 'The datacenter to use for the environment', {
+    required: true,
```

Comment:
I cannot find anything on using `required` and `conflicting` at the same time. This seems like undocumented behavior and we may just want to do the check ourselves sadly.

Reply:
Are you sure? It seems to work exactly as expected.

Reply:
I mean its not IN the docs anywhere on what the desired behavior of this should be. So either 
1. This feature is intended and undocumented.
2. This feature just happens to work but due to there being no test or documentation for it, could just change in the future with no warning.

Given we have 0 tests that test the whole CLI and can guarantee this behavior, it is generally better to avoid these kind of undocumented interactions.

Reply:
I did this all manually in the most recent changes.